apps/analysis/models.py

```diff
@@ -30,11 +30,63 @@ class Analysis(UserResource, ProjectEntityMixin):
     def __str__(self):
         return self.title
 
-    def clone_analysis(self):
+    def clone_analysis(self, title, end_date):
         analysis_cloned = copy.deepcopy(self)
+
+        def _get_clone_pillar(obj, analysis_cloned):
+            obj.pk = None
+            obj.client_id = None
+            obj.analysis = analysis_cloned
+            return obj
+
+        def _get_clone_pillar_statement(obj, analysis_pillar):
+            obj.pk = None
+            obj.client_id = None
+            obj.analysis_pillar = analysis_pillar
+            return obj
+
+        def _get_clone_statement_entry(obj, analytical_statement):
+            obj.pk = None
+            obj.client_id = None
+            obj.analytical_statement = analytical_statement
+            return obj
+
+        def _get_clone_discarded_entry(obj, analysis_pillar):
+            obj.pk = None
+            obj.analysis_pillar = analysis_pillar
+            return obj
+
         analysis_cloned.pk = None
-        analysis_cloned.title = f'{self.title} (cloned)'
+        analysis_cloned.client_id = None
+        analysis_cloned.title = title
+        analysis_cloned.end_date = end_date
```

Comment:
We need to store the `cloned_from` attribute to models (analysis, pillar, statement).